Integrations/Slack/Slack_test.py

```diff
@@ -334,80 +334,102 @@ def setup():
 
 
 @pytest.mark.asyncio
-async def test_get_slack_name(mocker):
+async def test_get_slack_name_user(mocker):
     from Slack import get_slack_name
 
     # Set
-
-    async def users_info(user):
-        if user != 'alexios':
-            return {'user': json.loads(USERS)[0]}
+    @asyncio.coroutine
+    def api_call(method: str, http_verb: str = 'POST', file: dict = None, params=None, json=None, data=None):
+        if method == 'users.info':
+            user = params['user']
+            if user != 'alexios':
+                return {'user': js.loads(USERS)[0]}
+        elif method == 'conversations.info':
+            return {'channel': js.loads(CONVERSATIONS)[0]}
         return None
 
-    async def conversations_info(channel):
-        if channel != 'lulz':
-            return {'channel': json.loads(CONVERSATIONS)[0]}
-
     mocker.patch.object(demisto, 'getIntegrationContext', side_effect=get_integration_context)
     mocker.patch.object(demisto, 'setIntegrationContext')
-    mocker.patch.object(slack.WebClient, 'users_info', side_effect=users_info)
-    mocker.patch.object(slack.WebClient, 'conversations_info', side_effect=conversations_info)
+    mocker.patch.object(slack.WebClient, 'api_call', side_effect=api_call)
 
     # Assert
 
     # User in integration context
     user_id = 'U012A3CDE'
     name = await get_slack_name(user_id, slack.WebClient)
     assert name == 'spengler'
-    assert slack.WebClient.users_info.call_count == 0
+    assert slack.WebClient.api_call.call_count == 0
 
     # User not in integration context
     unknown_user = 'USASSON'
     name = await get_slack_name(unknown_user, slack.WebClient)
     assert name == 'spengler'
-    assert slack.WebClient.users_info.call_count == 1
+    assert slack.WebClient.api_call.call_count == 1
 
     # User does not exist
     nonexisting_user = 'alexios'
     name = await get_slack_name(nonexisting_user, slack.WebClient)
     assert name == ''
-    assert slack.WebClient.users_info.call_count == 1
+    assert slack.WebClient.api_call.call_count == 1
+
+
+@pytest.mark.asyncio
+async def test_get_slack_name_channel(mocker):
+    from Slack import get_slack_name
+
+    # Set
+
+    @asyncio.coroutine
+    def api_call(method: str, http_verb: str = 'POST', file: dict = None, params=None, json=None, data=None):
```

Comment:
cant we define this function once and use it in all tests? or you want as many code lines as possible?

Reply:
It differs between some tests... and Yes.